frontend/www/js/omegaup/components/problem/creator/cases/AddPanel.test.ts

```diff
@@ -74,11 +78,111 @@ describe('AddPanel.vue', () => {
     const groups = store.state.casesStore.groups;
 
     expect(groups.length).toBe(1);
-    expect(groups[0].name).toBe('groupname');
+    expect(groups[0].name).toBe(testGroup);
     expect(groups[0].autoPoints).toBe(false);
     expect(groups[0].points).toBe(10);
     expect(groups[0].ungroupedCase).toBe(false);
+
+    const groupId = groups?.[0].groupID;
+
+    wrapper.setData({ tab: 'case' });
+    await Vue.nextTick();
+
+    const caseNameInput = wrapper.find('input[name="case-name"]');
+    const casePointsInput = wrapper.find('input[name="case-points"]');
+    const caseGroupName = wrapper.find('select[name="case-group"]');
+
+    await caseNameInput.setValue('groupedtestcase');
+    await casePointsInput.setValue(10);
+    await caseGroupName.setValue(groupId);
+
+    await wrapper.find('form').trigger('submit.prevent');
+
+    const groupedCases = store.state.casesStore.groups?.[0].cases;
```

Comment:
Same comment as above, here you already defined that the object exists, so you can omit the [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). Another thing you can do is [destructure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) the response:

```suggestion
    const [groupedCases] = store.state.casesStore.groups[0].cases;
```

This way you can directly access the element you're interested in without having to use index 0.